docs/investigations/secure-auth-resolvers.md

```diff
@@ -0,0 +1,80 @@
+# Limit access of GraphQL objects or fields for specific resource owners
+
+### Case 1:
+
+Every object should has access to read information only about itself.
+We should restrict access to other objects of the same type, for example: runtime musn't read other runtime configuraion.
+We have two kind of such resolvers:
+* `runtime(ID)` runtime should be able to only read about itself
+* `runtimes` - this data is limited for `RuntimeAgents` with `hasScopes` directive, so it's not our case.
+
+### Case 2:
+
+The runtime agent execute query `applicationsForRuntime` with `runtimeID` param.
+Runtime should't be able to call this query with different `runtimeID`. 
+In case of different ID, we should return  error such as `Access Denied`.
+
+### Case 3:
+
+RuntimeAgent want to read all Applications.
```

Comment:
```suggestion
RuntimeAgent wants to read all Applications.
```

Does the runtime want to read all defined applications or applications assigned to that particular runtime?